src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -202,7 +202,7 @@ public JedisPool setupNodeIfNotExist(HostAndPort node) {
       if (existingPool != null) return existingPool;
 
       JedisPool nodePool = new JedisPool(poolConfig, node.getHost(), node.getPort(),
-          connectionTimeout, soTimeout, password, 0, clientName, 
+          connectionTimeout, soTimeout, user, password, 0, clientName, 
```

Comment:
@tgrall Even if we had tests, we could not catch the issue because our current tests are based on `default` user and operations with default user and without user has same effect.

Of course we need more tests, for other classes as well. But those tests should be with user other that default and so we would require to create more Redis instances. As this needs some good amount time, I left it as a pending task for now.